server/odc-service/src/main/java/com/oceanbase/odc/service/resource/k8s/status/K8sDeploymentMatcher.java

```diff
@@ -0,0 +1,135 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.oceanbase.odc.service.resource.k8s.status;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.Validate;
+
+import com.oceanbase.odc.service.resource.ResourceState;
+import com.oceanbase.odc.service.resource.k8s.model.K8sDeployment;
+import com.oceanbase.odc.service.resource.k8s.model.K8sPod;
+
+import io.kubernetes.client.openapi.models.V1DeploymentSpec;
+import lombok.NonNull;
+import lombok.Setter;
+
+/**
+ * {@link K8sDeploymentMatcher}
+ *
+ * @author yh263208
+ * @date 2024-09-12 17:13
+ * @since ODC_release_4.3.2
+ */
+@Setter
+public class K8sDeploymentMatcher implements K8sResourceMatcher<K8sDeployment> {
+
+    private boolean matchesNullDeployment;
+    private boolean replicasEnough;
+    private boolean replicasNonEnough;
+    private boolean forAllPods;
+    private boolean forAnyPods;
+    private Set<ResourceState> podStatusIn = new HashSet<>();
+    private Integer minMatchesCountInHasPodStatuses = null;
+    private Set<ResourceState> hasPodStatuses = new HashSet<>();
+    private boolean matchesAllPodStatus;
+    private Set<ResourceState> podStatusNotIn = new HashSet<>();
+    private final ResourceState currentState;
+
+    public K8sDeploymentMatcher(@NonNull ResourceState currentState) {
+        this.currentState = currentState;
+    }
+
+    @Override
+    public boolean matches(K8sDeployment k8sResource) {
+        if (k8sResource == null && this.matchesNullDeployment) {
+            return true;
+        } else if (k8sResource == null) {
+            return false;
+        } else if (this.matchesNullDeployment) {
+            return false;
+        }
+        // k8sResource != null && !this.matchesNullDeployment
+        boolean matches = true;
+        V1DeploymentSpec spec = k8sResource.getSpec();
+        Validate.notNull(spec);
+        Validate.notNull(spec.getReplicas());
+        List<K8sPod> k8sPodList;
+        try {
+            k8sPodList = k8sResource.getK8sPods();
+        } catch (Exception e) {
+            return false;
+        }
+        if (CollectionUtils.isEmpty(k8sPodList)) {
+            return false;
+        }
+        int replicas = spec.getReplicas();
+        if (this.replicasEnough) {
+            matches &= (k8sPodList.size() >= replicas);
+        } else if (this.replicasNonEnough) {
+            matches &= (k8sPodList.size() < replicas);
+        }
+        if (this.forAllPods) {
+            matches &= k8sPodList.stream().allMatch(this::matchesPodStatus);
+            if (CollectionUtils.isNotEmpty(this.hasPodStatuses)) {
+                int matchesCount = 0;
+                for (ResourceState item : this.hasPodStatuses) {
```

Comment:
this function has too many logic case branch, suggest move sub logic out as a function to reduce logic depth

Reply:
done